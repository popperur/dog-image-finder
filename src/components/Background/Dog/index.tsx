import {
  BackLeg,
  BackLegFoot,
  DogBackTorso,
  DogContainer,
  FrontLeftFoot,
  FrontLeftLeg,
  FrontRightFoot,
  FrontRightLeg,
} from "components/Background/Dog/styles.tsx";
import DogHead from "components/Background/Dog/DogHead.tsx";
import DogTail from "components/Background/Dog/DogTail.tsx";

function Dog() {
  return (
    <DogContainer>
      <FrontLeftLeg>
        <FrontLeftFoot />
      </FrontLeftLeg>
      <DogBackTorso />
      <DogHead />
      <DogTail />
      <BackLeg>
        <BackLegFoot />
      </BackLeg>
      <FrontRightLeg>
        <FrontRightFoot />
      </FrontRightLeg>
    </DogContainer>
  );
}

export default Dog;
