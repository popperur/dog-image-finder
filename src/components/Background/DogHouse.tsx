import {
  DogHouseContainer,
  RoofLeft,
  RoofRight,
  RoofRightWall,
  WallLeft,
  WallRight,
  WallRightDoor,
} from "components/Background/styles.tsx";

function DogHouse() {
  return (
    <DogHouseContainer>
      <RoofRight>
        <RoofRightWall />
      </RoofRight>
      <WallLeft />
      <WallRight>
        <WallRightDoor />
      </WallRight>
      <RoofLeft />
    </DogHouseContainer>
  );
}

export default DogHouse;
