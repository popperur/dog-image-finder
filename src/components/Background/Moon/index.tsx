import {
  MoonCenter,
  MoonContainer,
} from "components/Background/Moon/styles.tsx";
import MoonCraters from "components/Background/Moon/MoonCraters";
import MoonGlows from "components/Background/Moon/MoonGlows";

function Moon() {
  return (
    <MoonContainer>
      <MoonGlows />
      <MoonCenter>
        <MoonCraters />
      </MoonCenter>
    </MoonContainer>
  );
}

export default Moon;
