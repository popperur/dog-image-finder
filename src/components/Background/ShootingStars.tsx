import {
  ShootingStar,
  ShootingStarContainer,
} from "components/Background/styles.tsx";
import useAnimationTimer from "hooks/useAnimationTimer.ts";

function ShootingStars() {
  const [keyIndex, animationDuration] = useAnimationTimer({
    initial: { duration: 4, delay: 6 },
    durationFn: () => 4 + Math.random() * 4,
    delayFn: () => 6 + Math.random() * 14,
  });

  return (
    <ShootingStarContainer>
      <ShootingStar
        $duration={animationDuration}
        key={`shooting-star-${keyIndex}`}
      />
    </ShootingStarContainer>
  );
}

export default ShootingStars;
