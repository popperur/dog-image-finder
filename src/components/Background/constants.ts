export const TOOLTIP_PAUSED =
  "Radio signal from the Moon! This incredible discovery could greatly enhance your experience in browsing for dogs!";
export const TOOLTIP_PLAYING = "Auuuuuu!";

// top, left, width, height, animationDelay
export const STARS = [
  [45, 66, 6, 6, 2],
  [70, 48, 5, 5, 4],
  [4, 44, 3, 3, 1],
  [23, 53, 1, 1, 2],
  [26, 93, 3, 3, 1.3],
  [46, 13, 4, 4, 2.3],
  [66, 34, 2, 2, 1.3],
  [36, 3, 3, 3, 3.3],
  [76, 54, 1, 1, 2.3],
  [56, 24, 3, 3, 1.3],
  [76, 25, 3, 3, 0.3],
  [56, 55, 1, 1, 2.31],
  [36, 35, 3, 3, 1.31],
  [86, 45, 2, 2, 2.31],
  [82, 85, 3, 3, 0.31],
  [81, 15, 1, 1, 1.31],
  [83, 35, 3, 3, 2.31],
  [23, 55, 2, 2, 4],
  [53, 50, 1, 1, 1.31],
  [83, 20, 1, 1, 1.51],
];

// top, left, width, height, opacity
export const MOON_CRATERS = [
  [23, 33, 17, 17, 0.5],
  [48, 10, 30, 30, 0.7],
  [90, 14, 20, 20, 0.4],
  [18, 63, 25, 25, 0.6],
  [33, 105, 10, 10, 0.8],
  [82, 55, 14, 14, 0.9],
  [122, 34, 20, 20, 0.6],
  [102, 84, 16, 16, 0.8],
  [63, 113, 20, 20, 0.9],
  [116, 119, 12, 12, 0.6],
];

// top, left, width, height, animationDuration, animationDelay
export const MOON_GLOWS = [
  [20, 30, 260, 260, 3, 0],
  [-20, -10, 340, 340, 3, 0.5],
];

// bottom, left, width, height
export const HILLS = [
  [183, 120, 20, 239],
  [259, 229, 28, 255],
  [211, 334, 28, 246],
];
