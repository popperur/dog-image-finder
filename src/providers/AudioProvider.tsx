import React, { useReducer, useRef } from "react";
import { audioReducer, AudioContext } from "context/AudioContext.ts";

interface AudioProviderProps {
  children: React.ReactNode;
}

function AudioProvider({ children }: AudioProviderProps) {
  const [state, dispatch] = useReducer(audioReducer, { isPlaying: false });
  const audioRef = useRef<HTMLAudioElement | null>(null);

  const play = () => {
    audioRef.current?.play();
    dispatch({ type: "PLAY" });
  };

  const pause = () => {
    audioRef.current?.pause();
    dispatch({ type: "PAUSE" });
  };

  return (
    <AudioContext.Provider value={{ state, play, pause }}>
      {children}
      <audio ref={audioRef} role="audio">
        <source src="/audio/moon_signal.mp3" type="audio/mpeg" />
      </audio>
    </AudioContext.Provider>
  );
}

export default AudioProvider;
