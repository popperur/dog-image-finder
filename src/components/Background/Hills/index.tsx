import {
  BackHill,
  HillContainer,
} from "components/Background/Hills/styles.tsx";
import Hill from "components/Background/Hills/Hill.tsx";
import { HILLS } from "components/Background/constants.ts";

function Hills() {
  const hillsProps = HILLS.map(([bottom, left, width, height]) => ({
    bottom,
    left,
    width,
    height,
  }));

  return (
    <HillContainer>
      {hillsProps.map((hillProps, index) => (
        <Hill key={index} {...hillProps} />
      ))}
      <BackHill />
    </HillContainer>
  );
}

export default Hills;
