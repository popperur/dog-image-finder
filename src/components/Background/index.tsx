import {
  BackgroundBlock,
  BackgroundContainer,
  Ground,
} from "components/Background/styles.tsx";
import DogHouse from "components/Background/DogHouse.tsx";
import ShootingStars from "components/Background/ShootingStars.tsx";
import Dog from "components/Background/Dog";
import Moon from "components/Background/Moon";
import Stars from "components/Background/Stars";
import Hills from "components/Background/Hills";

// The animated background is fetched from
// https://codepen.io/drexqq/pen/vvryLb/
// and reworked by randomizing animations and adding styled components.
// Original version: LeeSungHee <3

interface BackgroundProps {
  showElements: boolean;
}

function Background({ showElements }: BackgroundProps) {
  return (
    <BackgroundContainer>
      {showElements && (
        <BackgroundBlock>
          <Stars />
          <ShootingStars />
          <Moon />
          <Dog />
          <Hills />
          <DogHouse />
          <Ground />
        </BackgroundBlock>
      )}
    </BackgroundContainer>
  );
}

export default Background;
