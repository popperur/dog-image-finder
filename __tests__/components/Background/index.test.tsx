import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Background from "components/Background";
import AudioProvider from "providers/AudioProvider.tsx";

describe("Background component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <AudioProvider>
        <Background showElements={true} />
      </AudioProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
