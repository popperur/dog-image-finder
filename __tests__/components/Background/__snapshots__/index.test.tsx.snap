// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Background component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  overflow: hidden;
  height: 100vh;
  width: 100%;
  background-color: #2b333f;
}

.c1 {
  position: relative;
  top: 0;
  height: 91%;
  width: 90%;
  max-width: 820px;
  padding-top: 5%;
  margin: auto;
  transform: scale(1);
  transition: all 1s;
}

.c23 {
  position: absolute;
  top: 0;
  height: 51%;
  width: 55%;
}

.c24 {
  position: relative;
  top: -20%;
  left: -20%;
  height: 80px;
  width: 2px;
  background-color: #fff;
  box-shadow: 0 0 30px #fff;
  transform-origin: 10% 0;
  transform: rotate(130deg) scale(0);
  animation: giQQSW ease-in;
  animation-duration: 4s;
}

.c72 {
  position: absolute;
  bottom: 0;
  left: 27%;
  width: 344px;
  height: 271px;
  z-index: 90;
}

.c78 {
  position: absolute;
  top: 0;
  left: -16px;
  width: 253px;
  height: 141px;
  background-color: #1e4857;
  transform: skew(-33deg);
  border-right: 8px solid #4d4a55;
  border-bottom: 8px solid #4d4a55;
  border-bottom-right-radius: 5px;
  border-bottom-left-radius: 5px;
}

.c73 {
  position: absolute;
  top: 0;
  left: 37px;
  width: 292px;
  height: 140px;
  background-color: #1a3140;
  transform: skew(30deg);
  border-top-left-radius: 100%;
  border-top-right-radius: 12px;
  border-right: 8px solid #4d4a55;
  border-top: 8px solid #4d4a55;
}

.c74 {
  position: absolute;
  top: 0;
  left: 32px;
  width: 240px;
  height: 140px;
  background-color: #8c2c1f;
  transform: skew(0deg);
  border-bottom-right-radius: 16px;
}

.c75 {
  position: absolute;
  bottom: 0;
  width: 180px;
  height: 83%;
  background-color: #622224;
  border-top-left-radius: 5px;
}

.c76 {
  position: absolute;
  bottom: 0;
  right: -1px;
  width: 165px;
  height: 49%;
  background-color: #8c2c1f;
}

.c77 {
  position: absolute;
  top: 6%;
  left: 25px;
  height: 81%;
  width: 70%;
  background-color: #642220;
  border-radius: 50%;
  box-shadow: -14px 0 0 #5e2c2d inset;
}

.c79 {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 12px;
  background-color: #020303;
  border-radius: 20px;
  z-index: 110;
}

.c41 {
  position: absolute;
  bottom: 0;
  left: 20%;
  width: 195px;
  height: 253px;
  z-index: 100;
  transform: scale(0.9);
}

.c42 {
  position: absolute;
  right: 0;
  bottom: 0;
  width: 32px;
  height: 126px;
  background-color: #fa9028;
  border-top-left-radius: 50px;
  border-top-right-radius: 50px;
  overflow: hidden;
}

.c43 {
  position: absolute;
  right: -1px;
  bottom: -10px;
  width: 35px;
  height: 22px;
  background-color: #f1f1f1;
  border-radius: 50%;
  transform: skew(20deg) rotate(-22deg);
}

.c44 {
  position: absolute;
  bottom: 0;
  right: 70px;
  width: 100px;
  height: 60%;
  background-color: #ff9a29;
  border-top-left-radius: 180px;
  transform: skew(-10deg);
}

.c45 {
  position: absolute;
  right: 0;
  bottom: 0;
  width: 100px;
  height: 100%;
  background-color: #ff9a29;
  border-top-right-radius: 60px;
  border-top-left-radius: 60px;
  border-bottom-right-radius: 120px;
}

.c46 {
  position: absolute;
  top: -18px;
  width: 25px;
  height: 80px;
  background-color: #ff9a29;
}

.c47 {
  left: 0;
  border-top-right-radius: 40px;
}

.c48 {
  right: 0;
  border-top-left-radius: 40px;
}

.c49 {
  position: absolute;
  animation-duration: 2s;
  animation-timing-function: cubic-bezier(1, 0, 0, 1);
}

.c50 {
  top: 25px;
  right: -2px;
  width: 100%;
  height: 9px;
  transform: rotatex(9deg);
  animation-name: bylGws;
}

.c51 {
  position: absolute;
  width: 12px;
  height: 100%;
  background-color: #111;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

.c52 {
  left: 30px;
}

.c53 {
  left: 75px;
}

.c54 {
  position: absolute;
  top: 44px;
  width: 100%;
  height: 42px;
}

.c55 {
  right: -10px;
  width: 65px;
  height: 100%;
  background: #fff7ee;
  border-radius: 20px 20px 42px 38px;
  animation-name: jZKbdW;
}

.c56 {
  right: 0;
  top: 0;
  width: 17px;
  height: 14px;
  background: #000;
  border-radius: 0 0 0 38px;
  animation-name: cnrYMx;
}

.c57 {
  position: absolute;
  top: 92px;
  width: 100%;
  height: 10px;
  background-color: #c64f3c;
}

.c58 {
  position: absolute;
  left: 0;
  bottom: 0;
  width: 24.5px;
  height: 114px;
  background-color: #ff9a29;
}

.c59 {
  position: absolute;
}

.c62 {
  top: -38px;
  left: -28px;
  width: 12px;
  height: 70px;
  border-radius: 0 32px 0 0;
  border: 22px solid transparent;
  border-right-color: #ff9a29;
  transform: skew(0deg) rotate(-1deg) rotatey(1deg) skewy(-26deg);
}

.c61 {
  top: -23px;
  left: -27px;
  width: 14px;
  height: 9px;
  border-radius: 1000px;
  border: 12px solid transparent;
  border-right-color: #ff9a29;
  transform: skew(-7deg) rotate(-74deg) rotatey(1deg) skewy(-2deg) scale(2);
  animation: dPwbxO 150ms linear infinite;
}

.c60 {
  top: -34px;
  left: -15px;
  width: 0;
  height: 14px;
  border-radius: 0 32px 0 0;
  border: 15px solid transparent;
  border-right-color: #ff9a29;
}

.c63 {
  top: -43px;
  left: -27px;
  width: 20px;
  height: 24px;
  background-color: #f1f1f2;
  border-top-left-radius: 20px;
  border-bottom-left-radius: 20px;
  transform: skew(-2deg) rotate(-12deg);
  animation: hvbQKD 150ms linear infinite;
}

.c64 {
  position: absolute;
  left: 41px;
  bottom: 9px;
  width: 31px;
  height: 41px;
  background-color: #ff9a29;
  border: 4px solid transparent;
  border-top-color: #e48121;
  border-top-right-radius: 54px;
  transform: skew(10deg) rotate(41deg);
}

.c65 {
  position: absolute;
  left: 4px;
  bottom: -14px;
  width: 46px;
  height: 29px;
  background-color: #f1f1f1;
  border-radius: 50%;
  transform: skew(-15deg) skewx(18deg) rotate(-49deg);
}

.c66 {
  position: absolute;
  right: 39px;
  bottom: 0;
  width: 32px;
  height: 89px;
  background-color: #fa9028;
  border-top-left-radius: 50px;
  border-top-right-radius: 50px;
  overflow: hidden;
}

.c67 {
  position: absolute;
  left: -1px;
  bottom: -10px;
  width: 35px;
  height: 22px;
  background-color: #f1f1f1;
  border-radius: 50%;
  transform: skew(20deg) rotate(-22deg);
}

.c25 {
  position: absolute;
  top: 5%;
  left: -7%;
  width: 310px;
  height: 330px;
  opacity: 1;
}

.c28 {
  position: absolute;
  left: 80px;
  top: 70px;
  width: 160px;
  height: 160px;
  background-color: #d2e8e6;
  border-radius: 50%;
}

.c29 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 23px;
  left: 33px;
  width: 17px;
  height: 17px;
  opacity: 0.5;
}

.c30 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 48px;
  left: 10px;
  width: 30px;
  height: 30px;
  opacity: 0.7;
}

.c31 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 90px;
  left: 14px;
  width: 20px;
  height: 20px;
  opacity: 0.4;
}

.c32 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 18px;
  left: 63px;
  width: 25px;
  height: 25px;
  opacity: 0.6;
}

.c33 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 33px;
  left: 105px;
  width: 10px;
  height: 10px;
  opacity: 0.8;
}

.c34 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 82px;
  left: 55px;
  width: 14px;
  height: 14px;
  opacity: 0.9;
}

.c35 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 122px;
  left: 34px;
  width: 20px;
  height: 20px;
  opacity: 0.6;
}

.c36 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 102px;
  left: 84px;
  width: 16px;
  height: 16px;
  opacity: 0.8;
}

.c37 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 63px;
  left: 113px;
  width: 20px;
  height: 20px;
  opacity: 0.9;
}

.c38 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 116px;
  left: 119px;
  width: 12px;
  height: 12px;
  opacity: 0.6;
}

.c39 {
  position: relative;
  top: 54px;
  left: 17px;
  color: #888;
}

.c40 {
  cursor: pointer;
  width: 20px;
  animation: jDLGYP 2s infinite;
}

.c26 {
  position: absolute;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0.1;
  transition: opacity 0.3s;
  animation: xvTUm infinite;
  top: 20px;
  left: 30px;
  width: 260px;
  height: 260px;
  animation-duration: 3s;
  animation-delay: 0s;
}

.c27 {
  position: absolute;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0.1;
  transition: opacity 0.3s;
  animation: xvTUm infinite;
  top: -20px;
  left: -10px;
  width: 340px;
  height: 340px;
  animation-duration: 3s;
  animation-delay: 0.5s;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 42%;
}

.c3 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 45%;
  left: 66%;
  width: 6px;
  height: 6px;
  animation-delay: 2s;
}

.c4 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 70%;
  left: 48%;
  width: 5px;
  height: 5px;
  animation-delay: 4s;
}

.c5 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 4%;
  left: 44%;
  width: 3px;
  height: 3px;
  animation-delay: 1s;
}

.c6 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 23%;
  left: 53%;
  width: 1px;
  height: 1px;
  animation-delay: 2s;
}

.c7 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 26%;
  left: 93%;
  width: 3px;
  height: 3px;
  animation-delay: 1.3s;
}

.c8 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 46%;
  left: 73%;
  width: 4px;
  height: 4px;
  animation-delay: 2.3s;
}

.c9 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 66%;
  left: 34%;
  width: 2px;
  height: 2px;
  animation-delay: 1.3s;
}

.c10 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 36%;
  left: 83%;
  width: 3px;
  height: 3px;
  animation-delay: 3.3s;
}

.c11 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 76%;
  left: 54%;
  width: 1px;
  height: 1px;
  animation-delay: 2.3s;
}

.c12 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 56%;
  left: 64%;
  width: 3px;
  height: 3px;
  animation-delay: 1.3s;
}

.c13 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 76%;
  left: 75%;
  width: 3px;
  height: 3px;
  animation-delay: 0.3s;
}

.c14 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 56%;
  left: 55%;
  width: 1px;
  height: 1px;
  animation-delay: 2.31s;
}

.c15 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 36%;
  left: 35%;
  width: 3px;
  height: 3px;
  animation-delay: 1.31s;
}

.c16 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 86%;
  left: 45%;
  width: 2px;
  height: 2px;
  animation-delay: 2.31s;
}

.c17 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 82%;
  left: 85%;
  width: 3px;
  height: 3px;
  animation-delay: 0.31s;
}

.c18 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 81%;
  left: 75%;
  width: 1px;
  height: 1px;
  animation-delay: 1.31s;
}

.c19 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 83%;
  left: 35%;
  width: 3px;
  height: 3px;
  animation-delay: 2.31s;
}

.c20 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 23%;
  left: 55%;
  width: 2px;
  height: 2px;
  animation-delay: 4s;
}

.c21 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 53%;
  left: 50%;
  width: 1px;
  height: 1px;
  animation-delay: 1.31s;
}

.c22 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 83%;
  left: 80%;
  width: 1px;
  height: 1px;
  animation-delay: 1.51s;
}

.c68 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 183px;
  left: 120px;
  width: 20px;
  height: 239px;
}

.c69 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 259px;
  left: 229px;
  width: 28px;
  height: 255px;
}

.c70 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 211px;
  left: 334px;
  width: 28px;
  height: 246px;
}

.c71 {
  position: absolute;
  bottom: 0;
  right: 5%;
  height: 250px;
  width: 400px;
  background-color: #1f2731;
  border-top-left-radius: 500px;
  border-top-right-radius: 500px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
        <div
          class="c4"
        />
        <div
          class="c5"
        />
        <div
          class="c6"
        />
        <div
          class="c7"
        />
        <div
          class="c8"
        />
        <div
          class="c9"
        />
        <div
          class="c10"
        />
        <div
          class="c11"
        />
        <div
          class="c12"
        />
        <div
          class="c13"
        />
        <div
          class="c14"
        />
        <div
          class="c15"
        />
        <div
          class="c16"
        />
        <div
          class="c17"
        />
        <div
          class="c18"
        />
        <div
          class="c19"
        />
        <div
          class="c20"
        />
        <div
          class="c21"
        />
        <div
          class="c22"
        />
      </div>
      <div
        class="c23"
      >
        <div
          class="c24"
        />
      </div>
      <div
        class="c25"
      >
        <div
          class=""
        >
          <div
            class="c26"
          />
          <div
            class="c27"
          />
        </div>
        <div
          class="c28"
        >
          <div
            class=""
          >
            <div
              class="c29"
            />
            <div
              class="c30"
            />
            <div
              class="c31"
            />
            <div
              class="c32"
            />
            <div
              class="c33"
            />
            <div
              class="c34"
            />
            <div
              class="c35"
            />
            <div
              class="c36"
            />
            <div
              class="c37"
            />
            <div
              class="c38"
            />
            <div
              class="c39"
            >
              <div
                aria-label="toggle music"
                class="c40"
                role="button"
                tabindex="0"
              >
                <span
                  aria-label="SoundOutlinedIcon"
                  class="anticon anticon-sound"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="sound"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M625.9 115c-5.9 0-11.9 1.6-17.4 5.3L254 352H90c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h164l354.5 231.7c5.5 3.6 11.6 5.3 17.4 5.3 16.7 0 32.1-13.3 32.1-32.1V147.1c0-18.8-15.4-32.1-32.1-32.1zM586 803L293.4 611.7l-18-11.7H146V424h129.4l17.9-11.7L586 221v582zm348-327H806c-8.8 0-16 7.2-16 16v40c0 8.8 7.2 16 16 16h128c8.8 0 16-7.2 16-16v-40c0-8.8-7.2-16-16-16zm-41.9 261.8l-110.3-63.7a15.9 15.9 0 00-21.7 5.9l-19.9 34.5c-4.4 7.6-1.8 17.4 5.8 21.8L856.3 800a15.9 15.9 0 0021.7-5.9l19.9-34.5c4.4-7.6 1.7-17.4-5.8-21.8zM760 344a15.9 15.9 0 0021.7 5.9L892 286.2c7.6-4.4 10.2-14.2 5.8-21.8L878 230a15.9 15.9 0 00-21.7-5.9L746 287.8a15.99 15.99 0 00-5.8 21.8L760 344z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c41"
      >
        <div
          class="c42"
        >
          <div
            class="c43"
          />
        </div>
        <div
          class="c44"
        />
        <div
          class="c45"
        >
          <div
            class="c46 c47"
          />
          <div
            class="c46 c48"
          />
          <div
            class="c49 c50"
          >
            <div
              class="c51 c52"
            />
            <div
              class="c51 c53"
            />
          </div>
          <div
            class="c54"
          >
            <div
              class="c49 c55"
            >
              <div
                class="c49 c56"
              />
            </div>
          </div>
          <div
            class="c57"
          />
        </div>
        <div
          class="c58"
        >
          <div
            class="c59 c60"
          />
          <div
            class="c59 c61"
          />
          <div
            class="c59 c62"
          />
          <div
            class="c59 c63"
          />
        </div>
        <div
          class="c59 c64"
        >
          <div
            class="c59 c65"
          />
        </div>
        <div
          class="c66"
        >
          <div
            class="c67"
          />
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="c68"
        />
        <div
          class="c69"
        />
        <div
          class="c70"
        />
        <div
          class="c71"
        />
      </div>
      <div
        class="c72"
      >
        <div
          class="c73"
        >
          <div
            class="c74"
          />
        </div>
        <div
          class="c75"
        />
        <div
          class="c76"
        >
          <div
            class="c77"
          />
        </div>
        <div
          class="c78"
        />
      </div>
      <div
        class="c79"
      />
    </div>
  </div>
  <audio
    role="audio"
  >
    <source
      src="/audio/moon_signal.mp3"
      type="audio/mpeg"
    />
  </audio>
</DocumentFragment>
`;
